core/spec/lib/tasks/migrations/migrate_address_names_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+path = Spree::Core::Engine.root.join('lib/tasks/migrations/migrate_address_names.rake')
+
+RSpec.describe 'solidus:migrations:migrate_address_names' do
+  around do |example|
+    self.use_transactional_tests = false
+    ignored_columns = Spree::Address.ignored_columns
+    Spree::Address.ignored_columns = []
+
+    example.run
+
+  ensure
+    cleanup_address_data
+    self.use_transactional_tests = true
+    Spree::Address.ignored_columns = ignored_columns
+  end
+
+  describe 'up' do
+    include_context(
+      'rake',
+      task_path: path,
+      task_name: 'solidus:migrations:migrate_address_names:up'
+    )
+
+    context "when there are no records to migrate" do
+      it "simply exits" do
+        expect { task.invoke }.to output(
+          "Your DB contains 0 addresses that may be affected by this task.\n"
+        ).to_stdout
+      end
+    end
+
+    context "when there are records to migrate" do
+      let!(:complete_address) { create(:address, firstname: 'Jane', lastname: 'Smith') }
+      let!(:partial_address) { create(:address, firstname: nil, lastname: 'Doe') }
```

Comment:
I think it may be better to create an address with the partial name nullifying the last name because when we'll backport this into 2.11 we'll have a validation error since the first name is yet not optional on that version.

Reply:
Actually, this works fine in Solidus 2.11.x since we're actually [validating the presence](https://github.com/solidusio/solidus/blob/88ca1ceb3c7bd749aed78c769669e0d59ee4c7dd/core/app/models/spree/address.rb#L247) of `name` not `firstname`: `name` will be present since the `lastname` is set.